cloudferrylib/os/compute/host_aggregates.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2015 Mirantis Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and#
+# limitations under the License.
+
+
+from cloudferrylib.utils import utils as utl
+
+LOG = utl.get_log(__name__)
+
+
+class HostAggregates(object):
+    """
+    Host Aggregates are groupings of compute hosts within an availability
+    zone that contain some metadata about the host. A host can belong to
+    more than one aggregate, but only one AZ. Aggregates are used by flavors
+    that speciify metadata parameters scoped with the term
+    "aggregate_instance_extra_specs" to provide a means for special
+    scheduling of a insatnce that uses that flavor.
+
+    This class is available in CloudFerry as an attribute of of the
+    NovaCompute resource, `aggregates`, and maintains an instance attribute
+    containing a nova api client.
+    """
+
+    def __init__(self, api):
+        self.api = api
+
+    def list(self):
+        return self.api.aggregates.list()
+
+    def delete(self, aggregate):
+        return self.api.aggregates.delete(aggregate)
+
+    def deploy(self, src_host_aggregates):
+        dst_host_aggregates = self.list()
+        for src_aggregate in src_host_aggregates:
+            skip = False
+            for dst_aggregate in dst_host_aggregates:
+                if src_aggregate.name == dst_aggregate.name:
+                    if src_aggregate.metadata == dst_aggregate.metadata:
+                        LOG.info("skipping matching aggregate in "
+                                 "destination: %s", src_aggregate.name)
+                        skip = True
```

Comment:
Good suggestion. Although I believe what we want in this spot is `break`, then use the loop's optional else clause to perform the deploy.  This eliminates the need for any boolean governing when to proceed, and behaves identically. It also makes for much a more elegant look.